src/Catalog/DeleteCatalogItem.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using NuGet.Versioning;
+using VDS.RDF;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    public class DeleteCatalogItem : AppendOnlyCatalogItem
+    {
+        private string _id;
+        private string _version;
+        private DateTime _published;
+        private Guid _catalogItemId;
+
+        public DeleteCatalogItem(string id, string version, DateTime published)
+        {
+            _id = id;
+            _version = ParseVersion(version);
+            _published = published;
+
+            _catalogItemId = Guid.NewGuid();
+        }
+
+        private string ParseVersion(string version)
+        {
+            SemanticVersion semVer;
+            if (SemanticVersion.TryParse(version, out semVer))
+            {
+                return semVer.ToNormalizedString();
+            }
```

Comment:
is the point of this to normalize the version? If so this should be called `TryNormalize` or `Normalize`